### DISCUSSION TITLE
NavigationStack root view as @ViewBuilder?

### DISCUSSION THREAD

**Discussion by joshrl** [pointfreeco/swift-composable-architecture#3483] [Ideas]
I am working with the `NavigationStack(path:root:destination)` initializer and am noticing that the root view parameter is not marked as `@ViewBuilder`. Was this intentional? Seems like it would be beneficial  to have this behave the same way as the built-in `NavigationStack(path:root)` which does use `@ViewBuilder`. 

ie.: 

```swift
var body: some View {
    NavigationStack(path: $store.scope(state: \.path, action: \.path)) {
      // Can't build view here..
      // Wrapping in a `Group` or other container works here, but the following won't... 
      Text("FRED") 
      Text("WILMA") // Nope.
    } destination: { store in
      ...
    } 
```

I tested just changing the implementation as follows, and it does seem to work, but wondering if there is a rationale for not wanting that. Happy to open a PR if this was just an oversight, but just want to be sure I'm not missing something. 

```swift
  public init<State, Action, Destination: View, R>(
    path: Binding<Store<StackState<State>, StackAction<State, Action>>>,
    @ViewBuilder root: () -> R, // <-- Making this a view builder seems to work...
    @ViewBuilder destination: @escaping (Store<State, Action>) -> Destination,
    fileID: StaticString = #fileID,
    filePath: StaticString = #filePath,
    line: UInt = #line,
    column: UInt = #column
  )
```

Cheers,

Josh

**Comment by mbrandonw** [comment]
Hi @joshrl, this does seem to be just an oversight on our part. Would you be interested in preparing a PR to fix it?

**Reply by joshrl** [reply]
Done. #3493
